lib-src/rpc2/rpc2-src/pack_helper.c

```diff
@@ -0,0 +1,383 @@
+    if (len == 0)
+        *(RPC2_Unsigned *)(buf->buffer + 4) = 0;
+    else {
+      *(RPC2_Unsigned *)(buf->buffer + 4) = htonl(len);
+      memcpy(buf->buffer + 8, ptr, len);
+    }
+    buf->buffer += 8 + _PAD(len);
+    return 0;
+}
+
+
+int pack_encryptionKey(BUFFER *buf, char *ptr)
+{
+    if (buf->buffer + _PAD(RPC2_KEYSIZE) > buf->eob)
+        return -1;
+    memcpy(buf->buffer, ptr, RPC2_KEYSIZE);
+    buf->buffer += _PAD(RPC2_KEYSIZE);
+    return 0;
+}
+
+
+int pack_struct_CallCountEntry(BUFFER *buf, CallCountEntry *ptr, RPC2_Integer who)
+{
+    if (pack_string(buf, (char *)ptr->name))
+        return -1;
+    if (pack_int(buf, ptr->countent))
+        return -1;
+    if (pack_int(buf, ptr->countexit))
+        return -1;
+    if (pack_int(buf, ptr->tsec))
+        return -1;
+    if (pack_int(buf, ptr->tusec))
+        return -1;
+    if (pack_int(buf, ptr->counttime));
+        return -1;
+    return 0;
+}
+
+
+int unpack_struct_CallCountEntry(BUFFER *buf, CallCountEntry *ptr, RPC2_Integer who)
+{
+    if (unpack_string(buf, &(ptr->name), 0))
+        return -1;
+    if (unpack_int(buf, (RPC2_Integer *)&(ptr->countent)))
+        return -1;
+    if (unpack_int(buf, (RPC2_Integer *)&(ptr->countexit)))
+        return -1;
+    if (unpack_int(buf, (RPC2_Integer *)&(ptr->tsec)))
+        return -1;
+    if (unpack_int(buf, (RPC2_Integer *)&(ptr->tusec)))
+        return -1;
+    if (unpack_int(buf, (RPC2_Integer *)&(ptr->counttime)))
+        return -1;
+    return 0;
+}
+
+
+int pack_struct_MultiCallEntry(BUFFER *buf, MultiCallEntry *ptr, RPC2_Integer who)
+{
+    if (pack_string(buf, (char *)ptr->name))
+        return -1;
+    if (pack_int(buf, ptr->countent))
+        return -1;
+    if (pack_int(buf, ptr->countexit))
+        return -1;
+    if (pack_int(buf, ptr->tsec))
+        return -1;
+    if (pack_int(buf, ptr->tusec))
+        return -1;
+    if (pack_int(buf, ptr->counttime))
+        return -1;
+    return 0;
+}
+
+
+int unpack_struct_MultiCallEntry(BUFFER *buf, MultiCallEntry *ptr, RPC2_Integer who)
+{
+    if (unpack_string(buf, &(ptr->name), 0))
+        return -1;
+    if (unpack_int(buf, (RPC2_Integer *)&(ptr->countent)))
+        return -1;
+    if (unpack_int(buf, (RPC2_Integer *)&(ptr->countexit)))
+        return -1;
+    if (unpack_int(buf, (RPC2_Integer *)&(ptr->tsec)))
+        return -1;
+    if (unpack_int(buf, (RPC2_Integer *)&(ptr->tusec)))
+        return -1;
+    if (unpack_int(buf, (RPC2_Integer *)&(ptr->counttime)))
+        return -1;
+    return 0;
+}
+
+
+int pack_struct_MultiStubWork(BUFFER *buf, MultiStubWork *ptr, RPC2_Integer who)
+{
+    if (pack_int(buf, ptr->opengate))
+        return -1;
+    if (pack_int(buf, ptr->tsec))
+        return -1;
+    if (pack_int(buf, ptr->tusec))
+        return -1;
+    return 0;
+}
+
+
+int unpack_struct_MultiStubWork(BUFFER *buf, MultiStubWork *ptr, RPC2_Integer who)
+{
+    if (unpack_int(buf, (RPC2_Integer *)&(ptr->opengate)))
```

Comment:
I don't think these casts are necessary. The predefined stubs in `rp2gen/symtab.c` seem to already use `RPC2_*` as the types.